src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -198,8 +198,7 @@ class JvmtiVirtualThreadEventMark : public JvmtiEventMark {
 public:
   JvmtiVirtualThreadEventMark(JavaThread *thread) :
     JvmtiEventMark(thread) {
-    JvmtiThreadState* state = thread->jvmti_thread_state();
-    if (state != NULL && state->is_virtual()) {
+    if (thread->vthread() != NULL) {
```

Comment:
Yes, this condition can be false for platform threads.

Reply:
The [comment](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/runtime/javaThread.hpp#L88) suggests that `vthread()` cannot evaluate to NULL. Otherwise `Thread.currentThread()` would return null too.

I've experimented a little bit and found that `thread->vthread()` in fact can be NULL during initialization but then `thread->threadObj()` is also not yet initialized. E.g. in `Threads::initialize_java_lang_classes()` class events are generated before `create_initial_thread()` is reached. The constructor of `JvmtiClassEventMark` calls its parent class' constructor, that is `JvmtiVirtualThreadEventMark`,  with a not yet fully initialized initial thread.

I found that `jtreg:test/hotspot/jtreg:hotspot_serviceability` succeeds with `assert(thread->vthread() != NULL || thread->threadObj() == NULL, "");`
So you could unconditionally use `thread->vthread()`.